pkgs/development/cuda-modules/buildRedist/default.nix

```diff
@@ -345,39 +345,15 @@ extendMkDerivation {
 
       postFixup =
         postFixup
-        # Install the stub removal hook if requested.
-        #
-        # This must happen in postFixup since recordPropagatedDependencies overwrites these dependencies files during
-        # fixupPhase.
-        #
-        # NOTE:
-        #
-        #   We must add the hook to both the propagated-build-inputs and propagated-native-build-inputs to ensure the
-        #   the hook is used regardless of whether the CUDA dependency is in nativeBuildInputs or buildInputs,
-        #   respectively. We use the same selection logic with `__spliced` that `make-derivation.nix` uses for
-        #   `stdenv`.
-        #
-        #   As an example, if `cuda_cudart` (which has stubs) was in `nativeBuildInputs`, our hook should have been
-        #   sourced from propagated-build-inputs:
-        #
-        #   cuda12.8-cuda-samples> source: sourcing removeStubsFromRunpathHook.bash (hostOffset=-1) (targetOffset=0)
-        #
-        #   Likewise, if `cuda_cudart` was in `buildInputs`, our hook should have been sourced from
-        #   propagated-native-build-inputs:
-        #
-        #   cuda12.8-cuda-samples> source: sourcing removeStubsFromRunpathHook.bash (hostOffset=-1) (targetOffset=0)
-        #
-        #   This ensures that our hook is always running on the build platform and that the patchelf used is able to
-        #   execute on the build platform -- we cannot trust that users will always put packages in the correct
-        #   dependency arrays.
-        #
-        + optionalString finalAttrs.includeRemoveStubsFromRunpathHook ''
-          nixLog "installing stub removal runpath hook"
-          mkdir -p "''${!outputStubs:?}/nix-support"
-          printWords >>"''${!outputStubs:?}/nix-support/propagated-native-build-inputs" \
-            "${getDev removeStubsFromRunpathHook.__spliced.buildHost or removeStubsFromRunpathHook}"
-          printWords >>"''${!outputStubs:?}/nix-support/propagated-build-inputs" \
-            "${getDev removeStubsFromRunpathHook.__spliced.hostTarget or removeStubsFromRunpathHook}"
+        # Register removeStubsFromRunpathHook.
+        # This must happen in postFixup after recordPropagatedDependencies (fixupPhase).
+        + ''
+          if [[ -n "''${includeRemoveStubsFromRunpathHook:-}" ]] ; then
+            nixLog "installing stub removal runpath hook"
+            mkdir -p "''${!outputStubs:?}/nix-support"
+            printWords >>"''${!outputStubs:?}/nix-support/propagatedNativeBuildInputs" \
+              "${getDev removeStubsFromRunpathHook}"
```

Comment:
It's not enough to just record add the dependency to `propagated-native-build-inputs` -- that only works if we use stubs exclusively from `depsHostHost` or later (like `buildInputs`): https://github.com/ConnorBaker/nix-propagation-behavior/blob/1afbd58f2af1468d4564722b7180cc4d89967ef3/README.md?plain=1#L13-L18

Reply:
> propagated-build-inputs would mean that we'd have offsets (-1, 0) if the stubs are in nativeBuildInputs 

Yes, precisely, so it's within bounds of `findInputs`